src/inventory.cpp

```diff
@@ -29,6 +29,70 @@ bool invlet_wrapper::valid( const long invlet ) const
     return find( static_cast<char>( invlet ) ) != std::string::npos;
 }
 
+invlet_favorites::invlet_favorites( std::unordered_map<itype_id, std::string> map )
+{
+    for( const auto &p : map ) {
+        if( p.second.empty() ) {
+            // The map gradually accumulates empty lists; remove those here
+            continue;
+        }
+        invlets_by_id.insert( p );
+        for( char invlet : p.second ) {
+            uint8_t invlet_u = invlet;
+            if( !ids_by_invlet[invlet_u].empty() ) {
+                debugmsg( "Duplicate invlet: %s and %s both mapped to %c",
+                          ids_by_invlet[invlet_u], p.first, invlet );
+            }
+            ids_by_invlet[invlet_u] = p.first;
+        }
+    }
+}
+
+void invlet_favorites::set( char invlet, itype_id id )
```

Comment:
`itype_id` is basically a `std::string`, it should not be needed to copy it when calling this function (and the others as well). Take it as a const references instead. (Same applies to the `std::map` in the constructor above.)

Reply:
Good catch.